src/textual/widgets/_list_view.py

```diff
@@ -0,0 +1,145 @@
+    ]
+
+    index = reactive(0, always_update=True)
+
+    def __init__(
+        self,
+        *children: ListItem,
+        initial_index: int | None = 0,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+    ) -> None:
+        super().__init__(*children, name=name, id=id, classes=classes)
+        self.index = initial_index
+
+    @property
+    def highlighted_child(self) -> ListItem | None:
+        """Get the currently highlighted ListItem
+
+        Returns:
+            ListItem | None: The currently highlighted ListItem, or None if nothing is highlighted.
+        """
+        if self.index is None:
+            return None
+        elif 0 <= self.index < len(self.children):
+            return self.children[self.index]
+
+    def validate_index(self, index: int | None) -> int | None:
+        if not self.children or index is None:
+            return None
+        return self._clamp_index(index)
+
+    def _clamp_index(self, index: int) -> int:
+        """Clamp the index to a valid value given the current list of children"""
+        last_index = max(len(self.children) - 1, 0)
+        return clamp(index, 0, last_index)
+
+    def _is_valid_index(self, index: int | None) -> bool:
+        """Return True if the current index is valid given the current list of children"""
+        if index is None:
+            return False
+        return 0 <= index < len(self.children)
+
+    def watch_index(self, old_index: int, new_index: int) -> None:
+        if self._is_valid_index(old_index):
+            old_child = self.children[old_index]
+            old_child.highlighted = False
+        if self._is_valid_index(new_index):
+            new_child = self.children[new_index]
+            new_child.highlighted = True
+        else:
+            new_child = None
+
+        self._scroll_highlighted_region()
+        self.emit_no_wait(self.Highlighted(self, new_child))
+
+    async def append(self, item: ListItem) -> None:
+        """Append a new ListItem to the end of the ListView.
+
+        Args:
+            item (ListItem): The ListItem to append.
+        """
+        await self.mount(item)
+        if len(self) == 1:
+            self.index = 0
+        await self.emit(self.ChildrenUpdated(self, self.children))
+
+    async def clear(self) -> None:
+        """Clear all items from the ListView."""
+        await self.query("ListView > ListItem").remove()
+        await self.emit(self.ChildrenUpdated(self, self.children))
+
+    def action_select(self) -> None:
+        selected_child = self.highlighted_child
+        self.emit_no_wait(self.Selected(self, selected_child))
+
+    def on_list_item__child_clicked(self, event: ListItem._ChildClicked) -> None:
+        self.focus()
+        self.index = self.children.index(event.sender)
+        self.emit_no_wait(self.Selected(self, event.sender))
+
+    def key_up(self, event: events.Key) -> None:
```

Comment:
Yeah, this is where the handling is actually done. The bindings for up/down are dummies just to populate the footer. As far as I know when you use bindings you have no way to stop the event and so it was doing Textuals default scrolling behaviour when you pressed up and down.

I've hit a similar issue several times now - maybe I'm just missing something though. In my case I actually very rarely want the scrolling to occur when up/down is pressed.

Reply:
Ah, I think that is because the Widget class was using key events. I changed that in the tree pr to use bindings. You might find it will work with bindings now.